redo/builder.py

```diff
@@ -37,11 +37,13 @@ def start_stdin_log_reader(status, details, pretty, color,
     After this, be sure to run await_log_reader() before exiting.
     """
     global log_reader_pid
+    global stderr_fd
     r, w = os.pipe()    # main pipe to redo-log
     ar, aw = os.pipe()  # ack pipe from redo-log --ack-fd
     sys.stdout.flush()
     sys.stderr.flush()
     pid = os.fork()
+    stderr_fd = os.dup(2) # save our stderr for after the log pipe gets closed
```

Comment:
You should only do this in the parent process, ie where pid != 0.  The child process will terminate without needing to restore the stderr fd, so there is no need for it to keep a copy.

Reply:
I wasn't entirely sure about that, so I kept it in both.